source/portable/NetworkInterface/STM32/NetworkInterface.c

```diff
@@ -0,0 +1,708 @@
+
+void HAL_ETH_RxCpltCallback( ETH_HandleTypeDef * heth )
+{
+    static size_t uxMostRXDescsUsed = 0U;
+
+    const size_t uxRxUsed = heth->RxDescList.RxDescCnt;
+
+    if( uxMostRXDescsUsed < uxRxUsed )
+    {
+        uxMostRXDescsUsed = uxRxUsed;
+    }
+
+    if( xEMACTaskHandle != NULL )
+    {
+        BaseType_t xHigherPriorityTaskWoken = pdFALSE;
+        xTaskNotifyFromISR( xEMACTaskHandle, EMAC_IF_RX_EVENT, eSetBits, &xHigherPriorityTaskWoken );
+        portYIELD_FROM_ISR( xHigherPriorityTaskWoken );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+void HAL_ETH_TxCpltCallback( ETH_HandleTypeDef * heth )
+{
+    static size_t uxMostTXDescsUsed = 0U;
+
+    const size_t uxTxUsed = heth->TxDescList.BuffersInUse;
+
+    if( uxMostTXDescsUsed < uxTxUsed )
+	{
+		uxMostTXDescsUsed = uxTxUsed;
+	}
+
+    /* configASSERT( HAL_ETH_ReleaseTxPacket( heth ) == HAL_OK ); */
+
+    BaseType_t xHigherPriorityTaskWoken = pdFALSE;
+    xTaskNotifyFromISR( xEMACTaskHandle, EMAC_IF_TX_EVENT, eSetBits, &xHigherPriorityTaskWoken );
+    portYIELD_FROM_ISR( xHigherPriorityTaskWoken );
+}
+
+/*-----------------------------------------------------------*/
+
+void HAL_ETH_ErrorCallback( ETH_HandleTypeDef *heth )
+{
+    if( HAL_ETH_GetError( heth ) & HAL_ETH_ERROR_DMA )
+    {
+        if( HAL_ETH_GetDMAError( heth ) )
+        {
+            if( heth->gState == HAL_ETH_STATE_ERROR )
+            {
+                /* fatal bus error occurred */
+                /* F7 - ETH_DMASR_FBES | ETH_DMASR_TPS | ETH_DMASR_RPS */
+                /* H7 - ETH_DMACSR_FBE | ETH_DMACSR_TPS | ETH_DMACSR_RPS */
+                BaseType_t xHigherPriorityTaskWoken = pdFALSE;
+                xTaskNotifyFromISR( xEMACTaskHandle, EMAC_IF_ERR_EVENT, eSetBits, &xHigherPriorityTaskWoken );
+                portYIELD_FROM_ISR( xHigherPriorityTaskWoken );
+            }
+            else
+            {
+                /* F7 - ETH_DMASR_ETS | ETH_DMASR_RWTS | ETH_DMASR_RBUS | ETH_DMASR_AIS */
+                /* H7 - ETH_DMACSR_CDE | ETH_DMACSR_ETI | ETH_DMACSR_RWT | ETH_DMACSR_RBU | ETH_DMACSR_AIS */
+                /*if( ( HAL_ETH_GetDMAError(heth) & ETH_DMACSR_RBU ) == ETH_DMACSR_RBU )
+                {
+                    xSemaphoreGiveFromISR( xTxMutex );
+                }
+
+                if( ( HAL_ETH_GetDMAError(heth) & ETH_DMACSR_TBU ) == ETH_DMACSR_TBU )
+                {
+                    xSemaphoreGiveFromISR( xTxMutex );
+                }*/
+            }
+        }
+    }
+
+    #ifdef STM32H7
+        if( HAL_ETH_GetError( heth ) & HAL_ETH_ERROR_MAC )
+        {
+            if( HAL_ETH_GetMACError( heth ) )
+            {
+
+            }
+        }
+    #endif
+}
+
+/*-----------------------------------------------------------*/
+
+void HAL_ETH_RxAllocateCallback( uint8_t **buff )
+{
+	NetworkBufferDescriptor_t * pxBufferDescriptor = pxGetNetworkBufferWithDescriptor( ETH_RX_BUF_SIZE, pdMS_TO_TICKS( 0 ) );
```

Comment:
Are you sure that `HAL_ETH_RxAllocateCallback()` will only be called from a task-context?

And also, do you always use `BufferAllocation_2.c` ? I think we'll have to use that to make sure that uncached memory will be used by DMA.